### DISCUSSION TITLE
Commandbase and Async

### DISCUSSION THREAD

**Discussion by mikeclimbs23** [MarimerLLC/csla#2558] [Questions] [answered]
Within the DataPortal_Execute of command base I would like to call another command object.  Is there a way for me to call the other object asynchronously?

I'm having trouble getting the syntax correct for the override of protected override void DataPortal_Execute()

Thanks,

Mike

**Comment by rockfordlhotka** [comment, marked as answer]
You should be able to do something like this:

```c#
  [Execute]
  private async Task Execute()
  {
    var newCommand = await DataPortal.CreateAsync<SecondCommand>();
    await DataPortal.ExecuteAsync(newCommand);
  }
```
